src/solvers/thermal_operators.cpp

```diff
@@ -106,17 +106,17 @@ void DynamicConductionOperator::ImplicitSolve(const double dt, const mfem::Vecto
   x_       = 0.0;
 
   for (auto& bc : ess_bdr_) {
-    if (bc->scalar_coef != nullptr) {
-      bc->scalar_coef->SetTime(t);
+    if (std::holds_alternative<std::shared_ptr<mfem::Coefficient>>(bc.coef)) {
```

Comment:
Should this me an `ASSERT` as well?

Reply:
The "old" version looked like it was just ignoring `VectorCoefficient`s instead of crashing with an assertion failure, but if it's an invalid state (which it looks like it is because currently only `Coefficient`s  can be temperature boundary conditions), I can add an assertion.

Reply:
Yep, that was a bug. This should also be an assert.